src/Projection/AbstractProjection.php

```diff
@@ -133,10 +131,13 @@ protected function handleStreamWithHandlers(string $streamName, Iterator $events
         foreach ($events as $event) {
             /* @var Message $event */
             $this->position->inc($streamName);
+
             if (! isset($this->handlers[$event->messageName()])) {
                 continue;
             }
+
             $handler = $this->handlers[$event->messageName()];
+            $handler = Closure::bind($handler, $this->createHandlerContext($streamName));
```

Comment:
Perhaps we add stream name as third argument to the handler, but this is something I like to avoid.